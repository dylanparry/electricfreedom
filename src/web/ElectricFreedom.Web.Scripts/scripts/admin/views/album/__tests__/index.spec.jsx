import React from 'react';
import createContext from 'react-router-test-context';
import { mount } from 'enzyme';

import AlbumSwitch from '../index';
import AlbumAdd from '../add';
import AlbumEdit from '../edit';
import AlbumList from '../list';

const setup = (pathname = '/') =>
{
  const options = {
    context: createContext({
      location: { pathname },
    }),
    childContextTypes: {
      router: React.PropTypes.object,
    },
  };
  const wrapper = mount(<AlbumSwitch />, options);

  return {
    wrapper,
  };
};

describe('AlbumSwitch', () =>
{
  test('render', () =>
  {
    const { wrapper } = setup();

    expect(wrapper.exists()).toBe(true);
  });

  test('route "/artists/:artistId/albums/add" renders AlbumAdd', () =>
  {
    for (let i = 1; i <= 5; i += 1)
    {
      const { wrapper } = setup(`/artists/${i}/albums/add`);

      const albumAdd = wrapper.find(AlbumAdd);
      expect(albumAdd.exists()).toBe(true);
      expect(albumAdd.props().match.params.artistId).toBe(i.toString());
    }
  });

  test('route "/artists/:artistId/albums/:albumId" renders AlbumEdit', () =>
  {
    for (let i = 1; i <= 5; i += 1)
    {
      for (let j = 1; j <= 5; j += 1)
      {
        const { wrapper } = setup(`/artists/${i}/albums/${j}`);

        const albumEdit = wrapper.find(AlbumEdit);
        expect(albumEdit.exists()).toBe(true);
        expect(albumEdit.props().match.params.artistId).toBe(i.toString());
        expect(albumEdit.props().match.params.albumId).toBe(j.toString());
      }
    }
  });

  test('route "/artists/:artistId/albums" renders AlbumList', () =>
  {
    for (let i = 1; i <= 5; i += 1)
    {
      const { wrapper } = setup(`/artists/${i}/albums`);

      const albumList = wrapper.find(AlbumList);
      expect(albumList.exists()).toBe(true);
      expect(albumList.props().match.params.artistId).toBe(i.toString());
    }
  });
});
