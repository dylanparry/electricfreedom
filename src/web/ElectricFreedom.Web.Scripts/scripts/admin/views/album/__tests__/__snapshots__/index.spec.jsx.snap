// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumSwitch render 1`] = `
<albumSwitch>
  <Switch />
</albumSwitch>
`;
